lib/phoenix/endpoint.ex

```diff
@@ -909,13 +852,38 @@ defmodule Phoenix.Endpoint do
 
   """
   defmacro socket(path, module, opts \\ []) do
-    module = Macro.expand(module, %{__CALLER__ | function: {:__handler__, 2}})
+    module = Macro.expand(module, __CALLER__)
```

Comment:
We need to keep the %{function : ...} bit to avoid compile-time dependencies. If we need to pick another name, it can be `call/2`, as it is now a regular Plug.

Reply:
This is past the limit of my macro brain :) You're saying basically
```suggestion
    module = Macro.expand(module, %{__CALLER__ | function: {:call, 2}})
```

?

Reply:
Yes! Or `:socket_dispatch` instead of `:call` if we go down that route. The name doesn't matter.

Reply:
Done!